fdbclient/StorageServerInterface.h

```diff
@@ -741,6 +747,306 @@ struct ChangeFeedPopRequest {
 	}
 };
 
+// FDB storage checkpoint format.
+enum CheckpointFormat {
```

Comment:
IMO this should live in a separate header file

Reply:
I think the Checkpoint interface should be more generic and any storage engine specific details should live below the IKVS interface.

This can be done by making `CheckpointMetaData` store its storage engine specific details as
- A type enum, which could be 'CheckpointFormat' as it is now or could be 'KeyValueStoreType' and perhaps a 'subType'.
- An opaque string of serialized bytes, whose format is only known to storage engine (IKVS) implementations who support it.

Reply:
Moved to fdbclient/StorageCheckpoint.h.

Changed Checkpoint interface:

1. CheckpointFormat and CheckpointMetaData is used between clients and storage servers, KVS specific data is handled by corresponding components, e.g., fdbserver/RocksDBCheckpointUtils.actor.h, which are independent of storage server.
2. An opaque string is used in CheckpointMetaData for KVS-specific data.